cadc-access-control-server/src/main/java/ca/nrc/cadc/ac/server/ldap/LdapUserPersistence.java

```diff
@@ -155,37 +155,45 @@ public User addUser(User user)
     public User addUserRequest(UserRequest userRequest, final Principal ownerHttpPrincipal)
         throws UserNotFoundException, TransientException, AccessControlException, UserAlreadyExistsException
     {
-        LdapUserDAO userDAO = null;
         LdapConnections conns = new LdapConnections(this);
+        LdapUserDAO userDAO = null;
+        LdapGroupDAO groupDAO = null;
         User user = null;
+        Group group = null;
         try
         {
             // create the group to be associated with this userRequest
             userDAO = new LdapUserDAO(conns);
-            LdapGroupDAO groupDAO = new LdapGroupDAO(conns, userDAO);
+            groupDAO = new LdapGroupDAO(conns, userDAO);
             LocalAuthority localAuthority = new LocalAuthority();
             URI gmsServiceURI = localAuthority.getServiceURI(Standards.GMS_GROUPS_01.toString());
             GroupURI groupID = new GroupURI(gmsServiceURI, userRequest.getUser().getHttpPrincipal().getName());
-            Group group = new Group(groupID);
+            group = new Group(groupID);
             User groupOwner = userDAO.getAugmentedUser(ownerHttpPrincipal,  false);
             ObjectUtil.setField(group, groupOwner, "owner");
-
-            // ensure that there is no existing group with the same user name
-            boolean groupExists = checkIfGroupExists(group, groupDAO);
-            if (groupExists) {
-                throw new GroupAlreadyExistsException("Group " + group.getID().getName() + " already exists ");
-            }
```

Comment:
Why was this deleted?

Reply:
Not sure what you mean by 'don't see an update to build.gradle'. The version in build.gradle was updated. Perhaps you mean something else?

The bug was introduced by the deleted lines of code that you mentioned above. GroupDAO.addUserAssociatedGroup() checks whether the group already exists. Instead of checking for the existence of the group, we just create the UserRquest first and then create the group. By doing that, the checks for an already existing UserRequest (due to same userID or same email) fall into place.